### PR TITLE
Add support for ssh jump host.

### PR DESCRIPTION
**Pull request by sizeoftank**

### CODE REVIEW

**Comment by jeffpeiyt** [comment]
Thanks @sizeoftank for your contribution! Will review it more today and get back to you

File: src/main/java/io/parallec/core/commander/workflow/ssh/SshProvider.java [hunk 1]
```diff
@@ -107,36 +112,49 @@ public ResponseOnSingeRequest executeSshCommand() {
     }
 
     /**
-     * Start ssh session and obtain session.
```
Comment (jeffpeiyt) [inline src/main/java/io/parallec/core/commander/workflow/ssh/SshProvider.java]: Have some concerns on `sshJumpHostMeta` being type `SshMeta`.  SshMeta has around 10 attributes and I believe we just use username port and password , and this recursive pattern looks confusing to people.  

Could we just add the attribute directly to SshMeta, or create a new class just to wrap the attributes used?

File: src/main/java/io/parallec/core/commander/workflow/ssh/SshProvider.java [hunk 2]
```diff
@@ -151,6 +169,53 @@ public Session startSshSessionAndObtainSession() {
...
+            }
+
+            session.setConfig("StrictHostKeyChecking", "no");
+
+            session.connect();
+
+            logger.debug("The session has been established to {}@{}", jumpMeta.getUserName(),
+                    sshMeta.getSshJumpHost());
+
+            jsch = setJSchByMeta(jsch, sshMeta);
```
Comment (jeffpeiyt) [inline src/main/java/io/parallec/core/commander/workflow/ssh/SshProvider.java]: With lIne 177 and line 197 set twice, I understand the 2 metadata with the jump host info and this time the original metadata info being set into the jsch object.  however its readability could be improved. Suggest not to use the Class sshMeta to wrap the 3 attributes needed for the jump host.

Also seem with the `setJSchByMeta` function, we do not need the "=" assignment?   like 182 did not use the assignment

**Comment by jeffpeiyt** [comment]
@sizeoftank  thanks again for your great contribution and I am glad it passed all the tests.

Generally it looks good to me and I would definitely like this feature to be merged, just a couple of minor issues that it would be very nice if you can enhance it. Please check those 2  line comments. Here are some feedbacks:

I am sorry for these requests but all we want is to maintain high quality, clear and readable code with high coverage.

* Improve the coverage if you have time (optional). If not, I can help to add those tests later.
* Not to use SshMeta to wrap jump host attributes. Avoid recursive sshmeta, it is a bit confusing.

Would be great if you can enhance it, otherwise please submit a pull req to the  `dev` branch, I will continue to work on it.

File: src/main/java/io/parallec/core/bean/ssh/SshMeta.java [hunk 3]
```diff
@@ -58,6 +58,12 @@
     // ssh
     private int sshConnectionTimeoutMillis;
 
+    /** The ssh jump host name */
+    private String sshJumpHost;
+
+    /** The ssh jump host meta */
+    private SshMeta sshJumpHostMeta;
```
Comment (jeffpeiyt) [inline src/main/java/io/parallec/core/bean/ssh/SshMeta.java]: Have some concerns on `sshJumpHostMeta` being type `SshMeta`.  SshMeta has around 10 attributes and I believe we just use username port and password , and this recursive pattern looks confusing to people.  

Could we just add the attribute directly to SshMeta, or create a new class just to wrap the attributes used?

**Comment by sizeoftank** [comment]
Thanks for your response.
In the example we login with password and it also could login with key.
Now if we to avoid sshmeta recursives,  create another class wrap this?

**Comment by jeffpeiyt** [comment]
@sizeoftank Yes. Another class such as `sshJumpHostMeta` to wrap would be great. :-)

**Comment by jeffpeiyt** [comment]
Close this request for now as it is still WIP. Feel free to reopen it.

**Pull request closed without merging**